### DISCUSSION TITLE
Why make manifests is so slow?

### DISCUSSION THREAD

**Discussion by Peefy** [prometheus-operator/kube-prometheus#1902] [Q&A] [answered]
Hello.

I want to know why 'make manifests' is so slow. I haven't modified any jsonnet or libsonnet files. The following is my execution time:

```
> time jsonnet -J vendor example.jsonnet
jsonnet -J vendor example.jsonnet  249.29s user 1.21s system 98% cpu 4:15.09 total
```

**Comment by cateto** [comment, marked as answer]
https://github.com/prometheus-operator/kube-prometheus/blob/main/build.sh
```shell
./build example.jsonnet
```
I use this script above and It works fast for me 

### RESULT
```shell
$ time ./build.sh example.jsonnet
./build.sh example.jsonnet  2.94s user 0.15s system 97% cpu 3.183 total
```

**Comment by DimaVIII** [comment]
I was also shocked that compiling this few files takes horribly long. 
There are two jsonnet programs and I installed the wrong one~! go-jsonnet and jsonnet; The Go version is the right one.

Solution: `go install github.com/google/go-jsonnet/cmd/jsonnet@latest`
Info: https://github.com/google/go-jsonnet

The normal jsonnet is a terrible program, but google made a new pure go version of it, go-jsonnet. 
It's x1000 faster and it's a drop-in replacement. Remove jsonnet, install go-jsonnet.
`./build.sh kube.jsonnet  9.10s user 2.86s system 179% cpu 6.673 total`

Want to go faster? Try out sJsonnet:  https://www.databricks.com/blog/2018/10/12/writing-a-faster-jsonnet-compiler.html
